core/storage/email/gae_models.py

```diff
@@ -178,12 +178,11 @@ def create(
 
         email_model_instance.put()
 
-    def put(self, update_last_updated_time=True):
-        """Saves this SentEmailModel instance to the datastore."""
-        email_hash = self._generate_hash(
+    def _pre_put_hook(self):
+        """Operations to perform just before the model is `put` into storage."""
+        super(SentEmailModel, self)._pre_put_hook()
+        self.email_hash = self._generate_hash(
             self.recipient_id, self.subject, self.html_body)
-        self.email_hash = email_hash
-        super(SentEmailModel, self).put(update_last_updated_time)
```

Comment:
This changes the behaviour with respect to update_last_updated_time?

Reply:
Nope, since we aren't overriding put it'll receive the argument directly.